### ISSUE TITLE
Phase out serapeum dependency

### ISSUE THREAD

**Issue by ak-coram** [ak-coram/cl-duckdb#21]
This library currently prevents loading `cl-duckdb` with Clasp as it doesn't compile:
https://github.com/clasp-developers/clasp/issues/1365

Only `serapeum:mvlet*` and `serapeum:count-cpus` are currently used, which doesn't warrant the inclusion of such a large library dependency.

**Comment by ak-coram** [comment]
DuckDB seems to be using `std::thread::hardware_concurrency` to determine the number of threads, maybe this could be exposed via the C API.

**Comment by ak-coram** [comment]
@snunez1: I've replaced the single call we have to `mvlet*` with `let+`

> As a replacement for serapeum:count-cpus, perhaps cl-cpus?

This project was originally using cl-cpus before, but that was even less reliable compared to the serapeum implementation (by less reliable I mean it gave different results compared to the DuckDB defaults in more cases).

I've opened a PR where I simply open a temporary connection to DuckDB to determine their internal defaults when cl-duckdb is first loaded. This should be more future proof in case they decide to change the logic behind the defaults.
